examples/linear_model/plot_ridge_coeffs.py

```diff
@@ -1,89 +1,185 @@
+:math:`\\beta`.
+
+The regularized loss function is given by: :math:`\\mathcal{L}(X, y, \\beta) =
+\\| y - X \\beta \\|^{2}_{2} + \\alpha \\| \\beta \\|^{2}_{2}`
+
+where :math:`X` is the input data, :math:`y` is the target variable,
+:math:`\\beta` is the vector of coefficients associated with the features, and
+:math:`\\alpha` is the regularization strength.
+
+The regularized loss function aims to balance the trade-off between accurately
+predicting the training set and to prevent overfitting.
+
+In this regularized loss, the left-hand side (e.g. :math:`\\|y -
+X\\beta\\|^{2}_{2}`) measures the squared difference between the actual target
+variable, :math:`y`, and the predicted values. Minimizing this term alone could
+lead to overfitting, as the model may become too complex and sensitive to noise
+in the training data.
+
+To address overfitting, Ridge regularization adds a constraint, called a penalty
+term, (:math:`\\alpha \\| \\beta\\|^{2}_{2}`) to the loss function. This penalty
+term is the sum of the squares of the model's coefficients, multiplied by the
+regularization strength :math:`\\alpha`. By introducing this constraint, Ridge
+regularization discourages any single coefficient :math:`\\beta_{i}` from taking
+an excessively large value and encourages smaller and more evenly distributed
+coefficients. Higher values of :math:`\\alpha` force the coefficients towards
+zero. However, an excessively high :math:`\\alpha` can result in an underfit
+model that fails to capture important patterns in the data.
+
+Therefore, the regularized loss function combines the prediction accuracy term
+and the penalty term. By adjusting the regularization strength, practitioners
+can fine-tune the degree of constraint imposed on the weights, training a model
+capable of generalizing well to unseen data while avoiding overfitting.
 """
 
 # Author: Kornel Kielczewski -- <kornel.k@plusnet.pl>
 
-import matplotlib.pyplot as plt
+# %%
+# Purpose of this example
+# -----------------------
+# For the purpose of showing how Ridge regularization works, we will create a
+# non-noisy data set. Then we will train a regularized model on a range of
+# regularization strengths (:math:`\alpha`) and plot how the trained
+# coefficients and the mean squared error between those and the original values
+# behave as functions of the regularization strength.
+
+# %%
+# Ridge Regularization on a non-noisy data set
+# --------------------------------------------
+
+# %%
+# Creating a non-noisy data set
+###############################
+# We make a toy data set with 100 samples and 10 features, that's suitable to
+# detect regression. Out of the 10 features, 8 are informative and contribute to
+# the regression, while the remaining 2 features do not have any effect on the
+# target variable (their true coefficients are 0). Please note that in this
+# example the data is non-noisy, hence we can expect our regression model to
+# recover exactly the true coefficients w.
```

Comment:
I think that we only have to change the type of section. This should work.
I will make the proposed changes below.

```suggestion
# %%
# Purpose of this example
# -----------------------
# For the purpose of showing how Ridge regularization works, we will create a
# non-noisy data set. Then we will train a regularized model on a range of
# regularization strengths (:math:`\alpha`) and plot how the trained
# coefficients and the mean squared error between those and the original values
# behave as functions of the regularization strength.
#
# Ridge Regularization on a non-noisy data set
# --------------------------------------------
#
# Creating a non-noisy data set
# *****************************
# We make a toy data set with 100 samples and 10 features, that's suitable to
# detect regression. Out of the 10 features, 8 are informative and contribute to
# the regression, while the remaining 2 features do not have any effect on the
# target variable (their true coefficients are 0). Please note that in this
# example the data is non-noisy, hence we can expect our regression model to
# recover exactly the true coefficients w.
```